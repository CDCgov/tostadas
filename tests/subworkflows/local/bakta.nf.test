nextflow_workflow {

    name "Test Workflow RUN_BAKTA"
    script "subworkflows/local/bakta.nf"
    workflow "RUN_BAKTA"

    test("Should run without failures") {

        when {
            params {
                download_bakta_db = true
                bakta_db_type = 'light'
            }
            workflow {
                """
                input[0] = Channel.of([
                    [ batch_id: "batch_1", batch_tsv: file("${projectDir}/tests/subworkflows/local/batch_1_bacteria.tsv"), sample_id: "CP040557" ],
                    file("${projectDir}/assets/sample_fastas/bacteria/CP040557.fasta")
                ])
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

}
