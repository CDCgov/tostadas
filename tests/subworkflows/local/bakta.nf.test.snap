{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "batch_id": "batch_1",
                            "batch_tsv": "batch_1_bacteria.tsv:md5,25c21c0546d6beb4281768d216698a0c",
                            "sample_id": "CP040557"
                        },
                        "CP040557.gff3:md5,89a53e8515acb16f2454b40decb83df5"
                    ]
                ],
                "1": [
                    [
                        {
                            "batch_id": "batch_1",
                            "batch_tsv": "batch_1_bacteria.tsv:md5,25c21c0546d6beb4281768d216698a0c",
                            "sample_id": "CP040557"
                        },
                        "CP040557.fna:md5,3f29ef0f122a99db0129e8cec69cdeeb"
                    ]
                ],
                "fna": [
                    [
                        {
                            "batch_id": "batch_1",
                            "batch_tsv": "batch_1_bacteria.tsv:md5,25c21c0546d6beb4281768d216698a0c",
                            "sample_id": "CP040557"
                        },
                        "CP040557.fna:md5,3f29ef0f122a99db0129e8cec69cdeeb"
                    ]
                ],
                "gff": [
                    [
                        {
                            "batch_id": "batch_1",
                            "batch_tsv": "batch_1_bacteria.tsv:md5,25c21c0546d6beb4281768d216698a0c",
                            "sample_id": "CP040557"
                        },
                        "CP040557.gff3:md5,89a53e8515acb16f2454b40decb83df5"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-28T16:32:44.788889061"
    }
}